src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -504,6 +504,60 @@ namespace xclcpuemhal2 {
     sock = new unix_socket("EMULATION_SOCKETID");
   }
 
+  static size_t convert(const std::string& str)
+  {
+    return str.empty() ? 0 : std::stoul(str,0,0);
+  }
+
+  void CpuemShim::readXclbinXml(char* xmlfile, ssize_t xmlFileSize) {
+
```

Comment:
Please Dont add any readxml/readxclbin in driver. use common code which is in xclbin_parser.cpp.
Remaining code looks good.